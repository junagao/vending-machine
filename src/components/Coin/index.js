export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
