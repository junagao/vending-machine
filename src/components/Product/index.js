export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
