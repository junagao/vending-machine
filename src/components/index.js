export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
