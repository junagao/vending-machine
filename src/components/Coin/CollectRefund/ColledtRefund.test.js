import React from 'react';
import { shallow } from 'enzyme';
import { CollectRefund } from 'components';

describe('CollectRefund', () => {
  let wrapper;

  const machineCoinsAmountMock = 15;

  beforeEach(() => {
    wrapper = shallow(
      <CollectRefund
        onCollectCoinRefund={jest.fn()}
        machineCoinsAmount={machineCoinsAmountMock}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one div CollectRefundContainer element', () => {
    expect(wrapper.find('CollectRefundContainer').length).toEqual(1);
  });

  it('should render one button CollectRefundButton element of type `button` and text `Collect Refund`', () => {
    expect(wrapper.find('CollectRefundButton').length).toEqual(1);
    expect(wrapper.find('CollectRefundButton').prop('type')).toEqual('button');
    expect(
      wrapper
        .find('CollectRefundButton')
        .children()
        .text(),
    ).toEqual('Collect Refund');
  });
});
