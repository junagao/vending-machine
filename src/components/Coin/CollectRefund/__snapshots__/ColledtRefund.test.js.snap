// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectRefund should render correctly given the required props 1`] = `
<CollectRefundContainer>
  <CollectRefundButton
    onClick={[Function]}
    type="button"
  >
    Collect Refund
  </CollectRefundButton>
</CollectRefundContainer>
`;
